nemo/collections/asr/data/audio_to_text.py

```diff
@@ -288,6 +291,151 @@ def __init__(
         )
 
 
+@experimental
+class AudioToCharWithDursDataset(AudioToCharDataset):
+    """
+    Dataset that loads tensors via a json file containing paths to audio
+    files, transcripts, and durations (in seconds). Each new line is a
+    different sample. Example below:
+    {"audio_filepath": "/path/to/audio.wav", "text_filepath":
+    "/path/to/audio.txt", "duration": 23.147}
+    ...
+    {"audio_filepath": "/path/to/audio.wav", "text": "the
+    transcription", "offset": 301.75, "duration": 0.82, "utt":
+    "utterance_id", "ctm_utt": "en_4156", "side": "A"}
+
+    Additionally, user provides path to precomputed durations.
+
+    Args:
+        **kwargs: Passed to AudioToCharDataset constructor.
+        vocab_notation: Vocabulary to be used for text preprocessing.
+        vocab_punct: True if use punctuation for vocab.
+        vocab_spaces: True if use spaces for vocab.
+        vocab_stresses: True if use stresses for vocab.
+        durs_path: String path to pickled durations location.
+        rep: True if repeat text according to durs.
+    """
+
+    @property
+    def output_types(self) -> Optional[Dict[str, NeuralType]]:
+        """Returns definitions of module output ports."""
+        return {
+            'audio': NeuralType(
+                ('B', 'T'),
+                AudioSignal(freq=self._sample_rate)
+                if self is not None and hasattr(self, '_sample_rate')
+                else AudioSignal(),
+            ),
+            'audio_len': NeuralType(('B',), LengthsType()),
+            'text': NeuralType(('B', 'T'), LabelsType()),
+            'text_len': NeuralType(('B',), LengthsType()),
+            'durs': NeuralType(('B', 'T'), LengthsType()),
+        }
+
+    @staticmethod
+    def make_vocab(**kwargs):
+        notation = kwargs.get('vocab_notation', 'chars')
+        punct = kwargs.get('vocab_punct', True)
+        spaces = kwargs.get('vocab_spaces', False)
+        stresses = kwargs.get('vocab_stresses', False)
+        if notation == 'chars':
+            vocab = vocabs.Chars(punct=punct, spaces=spaces)
+        elif notation == 'phonemes':
+            vocab = vocabs.Phonemes(punct=punct, stresses=stresses, spaces=spaces)
+        else:
+            raise ValueError("Unsupported vocab type.")
+        return vocab
+
+    def __init__(self, **kwargs):
```

Comment:
Please have your function signature match your docstring. IE add `vocab_notation`, `vocab_punct`, etc.

Reply:
^ Please complete this (2/2)